request.go

```diff
@@ -62,10 +63,38 @@ func (r *Request) Encode(v interface{}) {
 	r.Header.Set("Content-Type", "application/json")
 }
 
-// Decode de-serialises the JSON body into the passed object.
+// EncodeAsProtobuf serialises the passed object as protobuf into the body
+func (r *Request) EncodeAsProtobuf(m proto.Message) {
+	out, err := proto.Marshal(m)
+	if err != nil {
+		r.err = terrors.Wrap(err, nil)
+		return
+	}
+
+	n, err := r.Write(out)
+	if err != nil {
+		r.err = terrors.Wrap(err, nil)
+		return
+	}
+	r.Header.Set("Content-Type", "application/x-protobuf")
```

Comment:
This smells like it could be wrong (or incompatible)?

When we encode, we set a Content-Type of `application/x-protobuf`.
When we decode we look for a range Content-Types, none of which include the one above.

Reply:
We [check for application/x-protobuf here](https://github.com/monzo/typhon/pull/127/files#diff-3c2feffcf57ffb6c596dc5608186c518252092b7017be5189d3dc12aa3c45cecR95-R100)